### PR TITLE
General update and cleaning

### PR DESCRIPTION
**Pull request by varyvol**
Basically:
- Update parent POM and plugin dependencies.
- Remove star imports.
- Remove Spotbugs errors.
- Stop using messages from other components.

It also changes the configuration that CI uses to validate, although it won't be executed on this PR builder because I lack permissions.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -73,41 +70,47 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>aws-java-sdk</artifactId>
-            <version>1.11.264</version>
+            <version>1.11.687</version>
```
Comment (alecharp) [inline pom.xml]: I'm wondering if such "meta-plugin" should be in the Jenkins BOM. 
But nothing to do here.

File: pom.xml [hunk 2]
```diff
@@ -73,41 +70,47 @@
...
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>structs</artifactId>
-            <version>1.2</version>
+            <version>1.20</version>
         </dependency>
     </dependencies>
 
     <repositories>
```
Comment (alecharp) [inline pom.xml]: You can also simply remove this, as the `https` url is defined on the parent pom.

File: src/main/java/hudson/plugins/s3/Entry.java [hunk 3]
```diff
@@ -113,11 +113,11 @@ public Entry(String bucket, String sourceFile, String excludedFile, String stora
 
     @Override
     public Descriptor<Entry> getDescriptor() {
-        return DESCRIPOR;
+        return DESCRIPTOR;
     }
 
     @Extension
-    public static final DescriptorImpl DESCRIPOR = new DescriptorImpl();
+    public static final DescriptorImpl DESCRIPTOR = new DescriptorImpl();
```
Comment (alecharp) [inline src/main/java/hudson/plugins/s3/Entry.java]: I *do* hope that no one was using the `DESCRIPOR` field directly before..

File: src/main/java/hudson/plugins/s3/MetadataPair.java [hunk 4]
```diff
@@ -26,11 +26,11 @@ public MetadataPair(String key, String value) {
     }
 
     public Descriptor<MetadataPair> getDescriptor() {
-        return DESCRIPOR;
+        return DESCRIPTOR;
     }
 
     @Extension
-    public final static DescriptorImpl DESCRIPOR = new DescriptorImpl();
+    public final static DescriptorImpl DESCRIPTOR = new DescriptorImpl();
```
Comment (alecharp) [inline src/main/java/hudson/plugins/s3/MetadataPair.java]: dito

File: src/main/java/hudson/plugins/s3/S3BucketPublisher.java [hunk 5]
```diff
@@ -6,9 +6,19 @@
...
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.Util;
+import hudson.model.AbstractProject;
+import hudson.model.Action;
+import hudson.model.Fingerprint;
+import hudson.model.Result;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.model. listeners.RunListener;
```
Comment (alecharp) [inline src/main/java/hudson/plugins/s3/S3BucketPublisher.java]: pretty sure this space shouldn't be here.

Comment (fcojfernandez) [inline src/main/java/hudson/plugins/s3/S3BucketPublisher.java]:
```suggestion
import hudson.model.listeners.RunListener;
```

File: src/main/java/hudson/plugins/s3/S3Profile.java [hunk 6]
```diff
@@ -296,6 +301,6 @@ public String toString() {
     }
 
     private ProxyConfiguration getProxy() {
-        return Jenkins.getActiveInstance().proxy;
+        return Jenkins.get().proxy;
```
Comment (alecharp) [inline src/main/java/hudson/plugins/s3/S3Profile.java]: I understand it's a `getActiveInstance()` to `get()` method switch, but accessing the `proxy` field is not really clean. 
Don't we have a cleaner way to get the `ProxyConfiguration`?

File: pom.xml (hunk 2 above)
Comment (varyvol) [reply pom.xml]: Where? I don't see it defined.

File: src/main/java/hudson/plugins/s3/Entry.java (hunk 3 above)
Comment (varyvol) [reply src/main/java/hudson/plugins/s3/Entry.java]: I don't think that's an expected use case, but I'll revert.

File: src/main/java/hudson/plugins/s3/S3Profile.java (hunk 6 above)
Comment (varyvol) [reply src/main/java/hudson/plugins/s3/S3Profile.java]: Not that I'm aware of. Neither the Jenkins nor the ProxyConfiguration classes have getters for it.

File: pom.xml (hunk 2 above)
Comment (alecharp) [reply pom.xml]: https://github.com/jenkinsci/pom/blob/2c3e0968fbd6bfed327bf33c9208b99cde78ca7e/pom.xml#L73

File: src/main/java/hudson/plugins/s3/S3Profile.java (hunk 6 above)
Comment (alecharp) [reply src/main/java/hudson/plugins/s3/S3Profile.java]: then it's fine for me.

File: src/main/java/hudson/plugins/s3/Entry.java (hunk 3 above)
Comment (alecharp) [reply src/main/java/hudson/plugins/s3/Entry.java]: No it's fine. Normally, scripts shouldn't be using it.

File: pom.xml (hunk 2 above)
Comment (fcojfernandez) [reply pom.xml]: But that's the pom parent for non-plugin components, right?

https://github.com/jenkinsci/pom/blob/2c3e0968fbd6bfed327bf33c9208b99cde78ca7e/pom.xml#L11-L13

Comment (alecharp) [reply pom.xml]: sorry, wrong link, same result:

https://github.com/jenkinsci/plugin-pom/blob/182757b442a0a97ce77a276434229301a8b1982b/pom.xml#L758

Comment (varyvol) [reply pom.xml]: I've checked removing it and the effective-pom is different. I'd prefer to leave it as is.

File: pom.xml [hunk 7]
```diff
@@ -13,10 +13,7 @@
     <url>https://wiki.jenkins.io/display/JENKINS/S3+Plugin</url>
 
     <properties>
-        <jenkins.version>1.625.3</jenkins.version>
```
Comment (alecharp) [inline pom.xml]: In general I prefer to keep the `jenkins.version` on each plugin pom, so that it's easier to know the target of the plugin

**Comment by alecharp** [comment]
Build is failing because the `Jenkinsfile` is not used, someone with admin access to fill a follow-up PR so the `Jenkinsfile` can be used? 

cc @batmat

File: pom.xml (hunk 7 above)
Comment (varyvol) [reply pom.xml]: TBH I have the same opinion but I wasn't sure of which the general approach is.

**Comment by fcojfernandez** [comment]
FYI @Jimilian , just in case you didn't notice this PR

Thanks in advance!

**Comment by varyvol** [comment]
@alecharp Jenkinsfile is now used and everything is working fine, so I'm merging this.

File: src/main/java/hudson/plugins/s3/callable/S3GzipCallable.java [hunk 8]
```diff
@@ -58,17 +66,17 @@ public void progressChanged(ProgressEvent event) {
...
 
-        try {
-            final InputStream gzipedStream = new FileInputStream(localFile);
+        try (final InputStream gzippedStream = new FileInputStream(localFile)) {
             final ObjectMetadata metadata = buildMetadata(file);
             metadata.setContentEncoding("gzip");
             metadata.setContentLength(localFile.length());
 
-            upload = Uploads.getInstance().startUploading(getTransferManager(), file, gzipedStream, getDest().bucketName, getDest().objectName, metadata);
+            upload = Uploads.getInstance().startUploading(getTransferManager(), file, gzippedStream, getDest().bucketName, getDest().objectName, metadata);
```
Comment (MMeent) [inline src/main/java/hudson/plugins/s3/callable/S3GzipCallable.java:79]: @varyvol @alecharp 

These select changes inadvertedly broke uploading using a gzipped stream, as the stream is automatically closed before the asynchronous upload task can complete, and causes JENKINS-60916.

This is being fixed in #127, so a merge + release would be appreciated. Currently, uploads using the latest release of the S3 plugin are quite a bit more expensive in bandwidth and S3-storage than they could be.

**Pull request merged**